cmd/kubeapps-apis/server/server.go

```diff
@@ -17,24 +17,74 @@ limitations under the License.
 package server
 
 import (
+	"context"
 	"fmt"
-	"html"
+	"net"
 	"net/http"
 
+	plugins "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/plugins/v1alpha1"
+	"github.com/soheilhy/cmux"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
 	log "k8s.io/klog/v2"
 )
 
 // Serve is the root command that is run when no other sub-commands are present.
 // It runs the gRPC service, registering the configured plugins.
 func Serve(port int, pluginDirs []string) {
-	// Stub echo http server
+
+	// Create the grpc server, register the standard services (reflection and our core service).
+	grpcSrv := grpc.NewServer()
+	reflection.Register(grpcSrv)
+	plugins.RegisterPluginsServiceServer(grpcSrv, &pluginsServer{})
+
+	// Create the http server, register our core service followed by any plugins.
 	listenAddr := fmt.Sprintf(":%d", port)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	gwArgs := gwHandlerArgs{
+		ctx:         ctx,
+		mux:         runtime.NewServeMux(),
+		addr:        listenAddr,
+		dialOptions: []grpc.DialOption{grpc.WithInsecure()},
```

Comment:
So we are not serving it under TLS (but once it is being routed by nginx, it "will" - as in the pinniped proxy), no?

Reply:
Correct.